rxandroid/src/main/java/rx/android/schedulers/HandlerScheduler.java

```diff
@@ -44,45 +45,49 @@ public Worker createWorker() {
     }
 
     static class HandlerWorker extends Worker {
-
-        final Handler handler;
-
-        private final CompositeSubscription compositeSubscription = new CompositeSubscription();
+        private final Handler handler;
+        /* Writes are guarded by 'this'. */
+        private volatile boolean unsubscribed;
 
         HandlerWorker(Handler handler) {
```

Comment:
I'll add a test to ensure multiple workers don't interfere with each other though.